pageserver/src/tenant/timeline/compaction.rs

```diff
@@ -928,6 +937,8 @@ impl Timeline {
             }
         }
 
+        drop(all_values_iter);
```

Comment:
there should be no difference whether to drop it or not b/c when we consume all values in the iterator, the iterator itself does not hold any resources. but it's okay to keep it there to make things explicit.

Reply:
There's a compiler error if you remove this on the `deltas_to_compact` still being borrowed 🤷‍♂️

Reply:
6f9ef9c596ef17e492f9a0b595215ee556886fd5